### PR TITLE
support macports for conf-perl modules

### PR DESCRIPTION
**Pull request by chetmurthy**
kartik provided this fix to support macports.  Hopefully it'll just go thru.

### CODE REVIEW

File: packages/conf-perl-ipc-system-simple/conf-perl-ipc-system-simple.2/opam [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+]
+depexts: [
+  ["libipc-system-simple-perl"] {os-family = "debian"}
+  ["perl-ipc-system-simple"] {os-distribution = "alpine"}
+  ["epel-release" "perl-IPC-System-Simple"] {os-distribution = "centos"}
+  ["perl-IPC-System-Simple"] {os-distribution = "ol" & os-version >= "8"}
+  ["perl-IPC-System-Simple"] {os-family = "suse"}
+  ["perl-IPC-System-Simple"] {os-distribution = "fedora"}
+  ["perl-ipc-system-simple"] {os-family = "arch"}
+  ["p5-ipc-system-simple"] {os-family = "macports"}
```
Comment (mseri) [inline packages/conf-perl-ipc-system-simple/conf-perl-ipc-system-simple.2/opam:22]:
```suggestion
  ["p5-ipc-system-simple"] {os-distribution = "macports" & os = "macos"}
```

File: packages/conf-perl-string-shellquote/conf-perl-string-shellquote.2/opam [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+]
+depexts: [
+  ["libstring-shellquote-perl"] {os-family = "debian"}
+  ["perl-string-shellquote"] {os-distribution = "alpine"}
+  ["perl-String-ShellQuote"] {os-distribution = "centos"}
+  ["perl-String-ShellQuote"] {os-distribution = "ol"}
+  ["perl-String-ShellQuote"] {os-family = "suse"}
+  ["perl-String-ShellQuote"] {os-family = "fedora"}
+  ["perl-string-shellquote"] {os-family = "arch"}
+  ["p5-string-shellquote"] {os-family = "macports"}
```
Comment (mseri) [inline packages/conf-perl-string-shellquote/conf-perl-string-shellquote.2/opam:22]:
```suggestion
  ["p5-string-shellquote"] {os-distribution = "macports" & os = "macos"}
```

Comment (chetmurthy) [reply packages/conf-perl-string-shellquote/conf-perl-string-shellquote.2/opam]: Applying your suggestion.

As long as you're looking, I have a workaround (from Blaisorblade) for the macos+homebrew problem: just disable testing on that platform.  It involves only modifying the opam file -- should I roll a fresh release, or just modify the opam file in opam-repository ?

Comment (mseri) [reply packages/conf-perl-string-shellquote/conf-perl-string-shellquote.2/opam]: If it is just about amending the test line on the opam file, that is fine to do here without a new release (as long as the tarball remains the same): you could do `{with-test & os != "macos"}`.

**Comment by mseri** [comment]
Thanks!

**Comment by k4rtik** [comment]
Hi @chetmurthy part of the patch that is missing is the following for the `conf-perl` package:
```
diff --git a/packages/conf-perl/conf-perl.1/opam b/packages/conf-perl/conf-perl.1/opam
index a55379b1c2..4a19336f4e 100644
--- a/packages/conf-perl/conf-perl.1/opam
+++ b/packages/conf-perl/conf-perl.1/opam
@@ -12,6 +12,7 @@ depexts: [
  ["perl"] {os-distribution = "arch"}
  ["perl-Pod-Html"] {os-distribution = "fedora"}
  ["perl-Pod-Html"] {os-distribution = "centos" & os-version >= "8"}
+  ["perl5"] {os-distribution = "macports" & os = "macos"}
]
synopsis: "Virtual package relying on perl"
description:
```

You may want to include it in your pending PR at https://github.com/ocaml/opam-repository/pull/20492 since this one is already merged. cc: @mseri 

It is important to match the perl version in `conf-perl` ("perl5") with that in the `conf-perl-*` module packages ("p5-*") for this to work properly with macports on macOS.

**Comment by chetmurthy** [comment]
Kartik, I added it to the other PR, but it caused something to blow up. So I took it out; I'll submit another PR once these both go thru.  One step at a time.  But thank you for pointing this out: I'll definitely fix it!  And again, thank you for doing this work on MacOS!

**Pull request merged**